File: plugin-azure-agent-base/src/main/java/jetbrains/buildServer/clouds/azure/WindowsCustomDataReader.java
package jetbrains.buildServer.clouds.azure;

import com.intellij.openapi.diagnostic.Logger;
import jetbrains.buildServer.agent.BuildAgentConfigurationEx;
import org.jetbrains.annotations.NotNull;

public class WindowsCustomDataReader extends AzureCustomDataReader {

    private static final Logger LOG = Logger.getInstance(WindowsCustomDataReader.class.getName());
    private static final String SYSTEM_DRIVE = System.getenv("SystemDrive");
    private static final String WINDOWS_CUSTOM_DATA_FILE = SYSTEM_DRIVE + "\\AzureData\\CustomData.bin";


    public WindowsCustomDataReader(@NotNull final BuildAgentConfigurationEx agentConfiguration,
                                   @NotNull final IdleShutdown idleShutdown,
                                   @NotNull final FileUtils fileUtils) {
        super(agentConfiguration, idleShutdown, fileUtils);
    }

    @Override
    protected String getCustomDataFileName() {
        return WINDOWS_CUSTOM_DATA_FILE;
    }

    @Override
    protected void parseCustomData(String customData) {
        // Process custom data
        try {
            processCustomData(customData);
        } catch (Exception e) {
            LOG.warnAndDebugDetails(String.format(UNABLE_TO_READ_CUSTOM_DATA_FILE, getCustomDataFileName()), e);
        }
    }
}

File: plugin-azure-agent/src/main/kotlin/jetbrains/buildServer/clouds/azure/WindowsCustomDataReader.kt


package jetbrains.buildServer.clouds.azure

import com.intellij.openapi.diagnostic.Logger
import jetbrains.buildServer.agent.BuildAgentConfigurationEx

class WindowsCustomDataReader(agentConfiguration: BuildAgentConfigurationEx,
                              fileUtils: FileUtils)
    : AzureCustomDataReader(agentConfiguration, fileUtils) {

    override val customDataFileName: String
        get() = WINDOWS_CUSTOM_DATA_FILE

    override fun parseCustomData(customData: String): MetadataReaderResult {
        // Process custom data
        return try {
            processCustomData(customData)
        } catch (e: Exception) {
            LOG.warnAndDebugDetails(String.format(UNABLE_TO_READ_CUSTOM_DATA_FILE, customDataFileName), e)
            MetadataReaderResult.SKIP
        }
    }

    companion object {
        private val LOG = Logger.getInstance(WindowsCustomDataReader::class.java.name)
        private val SYSTEM_DRIVE = System.getenv("SystemDrive")
        private val WINDOWS_CUSTOM_DATA_FILE = "$SYSTEM_DRIVE\\AzureData\\CustomData.bin"
    }
}
